dags/legacy_test/tests/pytorch/r2.3/common.libsonnet

```diff
@@ -0,0 +1,216 @@
+                  if re.match(libtpu_pattern, req)
+                ]
+                assert len(libtpu_matches) == 1, f'{len(libtpu_matches)} matches in {requirements} (pattern: `{libtpu_pattern}`)'
+                libtpu_date = libtpu_matches[0]
+                print('libtpu date:', libtpu_date)
+
+                ctc = cloud_tpu_client.Client(tpu=os.path.basename('$(TPU_NAME)'), zone=os.path.dirname('$(TPU_NAME)'))
+                ctc.wait_for_healthy()
+                ctc.configure_tpu_version(f'pytorch-2.3-dev{libtpu_date}', restart_type='always')
+                ctc.wait_for_healthy()
+              |||,
+            ],
+          },
+        },
+      },
+    },
+  },
+  Functional:: mixins.Functional {
+    schedule: '0 6 * * *',
+    tpuSettings+: {
+      preemptible: false,
+    },
+  },
+  Convergence:: mixins.Convergence,
+  PyTorchTpuVmMixin:: experimental.PyTorchTpuVmMixin + experimental.PjRt {
+    local config = self,
+
+    tpuSettings+: {
+      softwareVersion: 'tpu-ubuntu2204-base',
+      tpuVmPytorchSetup: |||
+        pip3 install -U setuptools
+        # `unattended-upgr` blocks us from installing apt dependencies
+        sudo systemctl stop unattended-upgrades
+        sudo apt-get -y update
+        sudo apt install -y libopenblas-base
+        # for huggingface tests
+        sudo apt install -y libsndfile-dev
+        # Install torchvision by pinned commit in PyTorch 2.3 release branch.
+        pip install torch==2.3.0 --index-url https://download.pytorch.org/whl/test/cpu
+        pip install --user --no-use-pep517 "git+https://github.com/pytorch/vision.git@2c127da8b5e2e8f44b50994c6cb931bcca267cfe"
+        pip install https://storage.googleapis.com/pytorch-xla-releases/wheels/tpuvm/torch_xla-2.3.0rc2-cp310-cp310-linux_x86_64.whl
+        pip install torch_xla[tpu] -f https://storage.googleapis.com/libtpu-releases/index.html
+        pip install pillow
+        git clone --depth=1 https://github.com/pytorch/pytorch.git
+        cd pytorch
+        git clone https://github.com/pytorch/xla.git
+      |||,
+    },
+    podTemplate+:: {
+      spec+: {
+        initContainerMap+:: {
+          'tpu-version': null,
+        },
+      },
+    },
+  },
+
+  datasetsVolume: volumes.PersistentVolumeSpec {
+    name: 'pytorch-datasets-claim',
+    mountPath: '/datasets',
+  },
+  GpuMixin:: {
+    local config = self,
+    imageTag+: '_cuda_12.1',
+
+    // TODO(wcromar): Merge TPU VM setup script with GPU entrypoint
+    tpuSettings+: {
+      tpuVmExports+: |||
+        export PJRT_DEVICE=CUDA
+      |||,
+    },
+
+    entrypoint: [
+      'bash',
+      '-cxue',
+      |||
+        export PATH=/usr/local/nvidia/bin${PATH:+:${PATH}}
+        export LD_LIBRARY_PATH=/usr/local/nvidia/lib64:/usr/local/nvidia/lib${LD_LIBRARY_PATH:+:${LD_LIBRARY_PATH}}
+
+        nvidia-smi
+        pip uninstall -y torch torchvision
+        pip install torch==2.3.0 --index-url https://download.pytorch.org/whl/test/cpu
```

Comment:
Sorry that this is duplicated lol

I really need to clean up these configs now...

Reply:
Thanks @will-cromar, removing these [lines](https://github.com/GoogleCloudPlatform/ml-auto-solutions/pull/198/files/5b327be8874d7bc34bf23e7acfb09c94e67af6c9#diff-be01cb501c71f2b4a7c669041385313192a6802fdda100de72e2e64f1b765895R142-R148) ok?

Reply:
You can skip installing `torch` and `torch_xla` since this code runs in our docker image that will have both. We will need to install `torchvision` and clone `xla`